metis/lib/folder_path_calculator.rb

```diff
@@ -17,7 +16,8 @@ def initialize(all_folders: nil, bucket:)
       @path_cache = {}
     end
 
-    def get_folder_path(folder)
+    def get_folder_path(folder_id)
+      folder = @folders_by_id[folder_id].first
```

Comment:
Basically this method / class wasn't doing any path caching or folder caching...oops. :facepalm: Hopefully these changes fix that, and paths are cached in memory and the "bucket-wide" folder list is only fetched once from the database. Not sure how I can measure or see differences locally, but seems like this should be fixed regardless.